leaf-ui/js/displayAnalysis.js

```diff
@@ -346,19 +351,48 @@ function addAnalysisConfig(config) {
     mainElement = document.getElementById(config.id);
     mainElement.querySelector('.config-elements').addEventListener('dblclick', function(){rename(this /** Config element */)});
     mainElement.querySelector('.config-elements').addEventListener('click', function(){switchConfigs(this.parentElement /** Config element */)});
-    mainElement.querySelector('.dropdown-button').addEventListener('click', function(){toggleDropdown(this.parentElement.parentElement /** Config element */)})
+    mainElement.querySelector('.dropdown-button').addEventListener('click', function(){toggleDropdown(this.parentElement.parentElement /** Config element */)});
+    mainElement.querySelector('.deleteconfig-button').addEventListener('click', function(){ 
+        try{removeConfiguration(this.parentElement.parentElement)/** Config element */} 
+        catch(e){console.error(e); alert(e);}
+    });
+
+    // Whenever first or loaded configs are added, update the highest position 
+    // Check if configuration added has higher initialPosition than current highest position, prevents naming duplicates
+    if(config.initialPosition > highestPosition) {
+        highestPosition = config.initialPosition;
+        console.log(highestPosition);
+    }
 }
 
 /**
  * Removes an analysis configuration from the analysisMap and UI sidebar
  * TODO implement with a button
+ * @param {HTML Element} configElement
  */
-function removeConfiguration() {
+function removeConfiguration(configElement) {
+    console.log(currAnalysisConfig);
```

Comment:
I think you can also delete this console.log unless you have a reason to leave it in!